controller/src/lib.rs

```diff
@@ -25,12 +25,22 @@ pub async fn run_scheduler(nats: TypedNats, state: StateHandle) -> NeverResult {
     let mut schedule_request_sub = nats.subscribe(ScheduleRequest::subscribe_subject()).await?;
     tracing::info!("Subscribed to spawn requests.");
 
-    while let Some(schedule_request) = schedule_request_sub.next().await {
-        tracing::info!(spawn_request=?schedule_request.value, "Got spawn request");
-        let result = match scheduler.schedule(&schedule_request.value.cluster, Utc::now()) {
+    while let Some(
+        ref msg @ MessageWithResponseHandle {
```

Comment:
This will end the scheduler loop if we get a schedule request for another type of resource, I don't think that's what you intended?